File: extensions/md2gemini/convert.py
#!/usr/bin/env python3
"""Convert a markdown command (YAML frontmatter + body) to a Gemini CLI TOML command.

Reads markdown from stdin, writes TOML to stdout.
Mapping:
  frontmatter `description` -> TOML `description`
  markdown body            -> TOML `prompt` (triple-quoted)
  Claude placeholders $ARGUMENTS / $1.. -> Gemini {{args}}
Unknown frontmatter keys are dropped.
"""
import sys
import re


def split_frontmatter(text):
    if text.startswith("---"):
        end = text.find("\n---", 3)
        if end != -1:
            fm = text[3:end].strip("\n")
            body = text[end + 4:].lstrip("\n")
            return fm, body
    return "", text


def get_description(fm):
    for line in fm.splitlines():
        m = re.match(r"\s*description\s*:\s*(.+?)\s*$", line)
        if m:
            return m.group(1).strip().strip("\"'")
    return ""


def toml_escape_basic(s):
    return s.replace("\\", "\\\\").replace('"', '\\"')


def main():
    text = sys.stdin.read()
    fm, body = split_frontmatter(text)
    description = get_description(fm)
    prompt = re.sub(r"\$ARGUMENTS\b", "{{args}}", body)
    prompt = re.sub(r"\$\d+", "{{args}}", prompt)

    out = []
    if description:
        out.append('description = "%s"' % toml_escape_basic(description))
    fence = '"""'
    if fence in prompt:
        fence = "'''"
    out.append("prompt = %s\n%s\n%s" % (fence, prompt.rstrip("\n"), fence))
    sys.stdout.write("\n".join(out) + "\n")


if __name__ == "__main__":
    main()

File: extensions/gemini-commands/convert.js
#!/usr/bin/env node
// Convert a markdown command (YAML frontmatter + body) to a Gemini CLI TOML command.
//
// Reads markdown from stdin, writes TOML to stdout.
// Mapping:
//   frontmatter `description` -> TOML `description`
//   markdown body            -> TOML `prompt` (triple-quoted)
//   Claude placeholders $ARGUMENTS / $1.. -> Gemini {{args}}
// Unknown frontmatter keys are dropped.

function splitFrontmatter(text) {
  if (text.startsWith("---")) {
    const end = text.indexOf("\n---", 3);
    if (end !== -1) {
      const fm = text.slice(3, end).replace(/^\n+|\n+$/g, "");
      const body = text.slice(end + 4).replace(/^\n+/, "");
      return [fm, body];
    }
  }
  return ["", text];
}

function getDescription(fm) {
  for (const line of fm.split("\n")) {
    const m = line.match(/^\s*description\s*:\s*(.+?)\s*$/);
    if (m) return m[1].trim().replace(/^["']|["']$/g, "");
  }
  return "";
}

function tomlEscapeBasic(s) {
  return s.replace(/\\/g, "\\\\").replace(/"/g, '\\"');
}

function readStdin() {
  return new Promise((resolve) => {
    let data = "";
    process.stdin.setEncoding("utf8");
    process.stdin.on("data", (chunk) => (data += chunk));
    process.stdin.on("end", () => resolve(data));
  });
}

(async () => {
  const text = await readStdin();
  const [fm, body] = splitFrontmatter(text);
  const description = getDescription(fm);
  let prompt = body.replace(/\$ARGUMENTS\b/g, "{{args}}").replace(/\$\d+/g, "{{args}}");

  const out = [];
  if (description) out.push(`description = "${tomlEscapeBasic(description)}"`);
  let fence = '"""';
  if (prompt.includes(fence)) fence = "'''";
  out.push(`prompt = ${fence}\n${prompt.replace(/\n+$/, "")}\n${fence}`);
  process.stdout.write(out.join("\n") + "\n");
})();
